src/lib/challenge.ts

```diff
@@ -48,3 +52,26 @@ export async function generate (size: number = 32): Promise<string> {
     throw error
   }
 }
+
+/**
+ * Helper function to validate signatures using public key
+ * @param challenge UTF-8 challenge string
+ * @param signature Base64 sign string
+ * @param publicKey Base64 RSA/EC:secp256k1 key string or KeyObject for verification
+ */
+export function verifySign (
+  challenge: string,
+  signature: string,
+  publicKey: string | crypto.KeyObject): boolean {
+  try {
+    // Hash using SHA256
+    const verifier: crypto.Verify = crypto.createVerify('SHA256')
```

Comment:
there is no need to create verifier for every call of `verifySign` it will slow down verification, make `verifier` module's constant instead.

Reply:
@eoln you need a different verifier for each signature. Signing many challenges with the same one doesn't work.